Upgrade to Hibernate ORM 6.4, Hibernate Search 7.0, Hibernate Reactive 2.2

yrodiere opened the pull request:
~Creating as draft for now, because:~

* ~We need an ORM 6.4.0.CR2/Final.~ => Done
* ~We still need a HR release and need to upgrade to that, because the current version doesn't work with ORM 6.4.~ => Done
* ~We may need a Hibernate Search 7.0.0.CR2/Final, but just for good measure as it seems to work as is.~ => Done

Fixes #37184
Fixes #36363
Fixes #35310
Fixes #33895
Fixes #32532
Fixes #33740

yrodiere committed:
Upgrade to Hibernate ORM 6.3.1.Final and Hibernate Reactive 2.1.0.Final

Co-authored-by: Stéphane Épardaud <stef@epardaud.fr>

yrodiere committed:
Upgrade to Hibernate ORM 6.4.0.Final and Hibernate Reactive 2.2.0.Final

yrodiere committed:
Fix invalid use of "!= null"

This used to be translated to SQL as "is not null", but will be
translated as "<> null" in newer versions of Hibernate ORM, which
obviously cannot work.

See https://github.com/hibernate/hibernate-orm/blob/6.3/migration-guide.adoc#hql-null-literal-comparison

yrodiere committed:
Upgrade to Hibernate Search 7.0.0.CR2

Necessary for official compatibility with ORM 6.4.

yrodiere committed:
Remove unused class HibernateSearchOutboxPollingClasses

Follows up on 0005466c611ba51a7159cab36b6314e82a65a426 which removed the
need for this workaround.

yrodiere committed:
Import the Hibernate Search BOM into the Quarkus BOM

yrodiere committed:
Fix invalid links in Hibernate Search extension documentation

yrodiere committed:
Rename extension quarkus-hibernate-search-orm-coordination-outbox-polling to quarkus-hibernate-search-orm-outbox-polling

To align on Hibernate Search renaming the module
hibernate-search-orm-coordination-outbox-polling to hibernate-search-orm-outbox-polling

yrodiere committed:
Upgrade to quarkus-local-cache 0.3.0

FroMage commented:
Excellent!

yrodiere committed:
Align Hibernate Search configuration defaults on Hibernate Search 7

FroMage commented:
The 6.4.0-SNAPSHOT branch (after CR1) wants one extra method in `CustomMetadataBuilderContributor.HardcodedSuffixFunction`

yrodiere commented:
@beikov [this change](https://github.com/hibernate/hibernate-orm/commit/bfb9fcc0f6e2e273447af61818126e9816d4b50f) seems to introduce a backwards-incompatible API change for implementations of `AbstractSqmSelfRenderingFunctionDescriptor`: there is no default implementation for `render(SqlAppender sqlAppender, List<? extends SqlAstNode> sqlAstArguments, ReturnableType<?> returnType, SqlAstTranslator<?> walker)`. Should we fix this in ORM?

yrodiere commented:
Ah, all things SQM are considered incubating apparently:

https://docs.jboss.org/hibernate/orm/6.4/incubating/incubating.txt

```
org.hibernate.query.sqm.*
```

So... breaking changes are fair, please ignore me.

yrodiere commented:
I added the upgrade to Hibernate ORM 6.4.0.Final, Hibernate Reactive 2.2.0.Final and Hibernate Search 7.0.0.CR2.

The tests should pass now, though I can't run them locally right now, so I marked as ready for review and will see what CI has to say about it.

yrodiere commented:
Not sure what's going on, but CI skipped almost all jobs... I rebased and force-pushed, let's see how it goes.

yrodiere committed:
Avoid reusing a non-empty persistence context across transaction in tests

Because:

1. It's a dodgy practice, at best.
2. It triggers some complex behavior in Hibernate ORM related to
   optimistic locking, and we really don't want to have to deal with
   that in unrelated tests.

yrodiere committed:
Fix assertions in tests related to Hibernate ORM FormatMappers

yrodiere committed:
Fix weird transaction in tests related to Hibernate ORM FormatMappers

When the expected transaction was thrown, the entity manager was closed
and the transaction commit wasn't completely done (the flush failed on
pre-commit), so we ended up in a strange state that happens to affect
other tests, causing problems when we upgrade Hibernate ORM.

yrodiere commented:
I tried to fix what I noticed in the previous CI run, let's try again...

@marko-bekhta you might be interested by commits 2, 3 and 4. Turns out the tests you added for FormatMappers were more complex than I thought when I reviewed them xD

yrodiere commented:
I think I addressed most failures, but the one in Panache Hibernate Reactive ITs puzzles me... I wonder if, maybe, Hibernate Reactive now parses queries asynchronously, resulting in some `catch` block in Panache no longer working (because we would need to use `Uni#onFailure` instead)?

@DavideD @FroMage Do you think you could have a look?

FTR, the failure:

```
2023-11-24T18:53:35.4762686Z [ERROR] Tests run: 23, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 1.961 s <<< FAILURE! -- in io.quarkus.it.panache.reactive.PanacheFunctionalityTest
2023-11-24T18:53:35.4764946Z [ERROR] io.quarkus.it.panache.reactive.PanacheFunctionalityTest.testPanacheFunctionality -- Time elapsed: 0.212 s <<< FAILURE!
2023-11-24T18:53:35.4766256Z java.lang.AssertionError: 
2023-11-24T18:53:35.4766734Z 1 expectation failed.
2023-11-24T18:53:35.4767266Z Response body doesn't match expectation.
2023-11-24T18:53:35.4767800Z Expected: is "OK"
2023-11-24T18:53:35.4889395Z   Actual: {"details":"Error id 247d4f49-eef7-4d6e-a987-785d741a0b5b-1, java.lang.IllegalArgumentException: org.hibernate.query.SyntaxException: At 1:70 and token '<EOF>', mismatched input '<EOF>', expecting one of the following tokens: '=', '.' [UPDATE io.quarkus.it.panache.reactive.Person SET Person.updateAllNames]","stack":"java.lang.IllegalArgumentException: org.hibernate.query.SyntaxException: At 1:70 and token '<EOF>', mismatched input '<EOF>', expecting one of the following tokens: '=', '.' [UPDATE io.quarkus.it.panache.reactive.Person SET Person.updateAllNames]\n\tat org.hibernate.internal.ExceptionConverterImpl.convert(ExceptionConverterImpl.java:143)\n\tat org.hibernate.internal.ExceptionConverterImpl.convert(ExceptionConverterImpl.java:167)\n\tat org.hibernate.internal.ExceptionConverterImpl.convert(ExceptionConverterImpl.java:173)\n\tat org.hibernate.reactive.session.impl.ReactiveExceptionConverter.convert(ReactiveExceptionConverter.java:30)\n\tat org.hibernate.reactive.session.impl.ReactiveSessionImpl.createReactiveQuery(ReactiveSessionImpl.java:377)\n\tat org.hibernate.reactive.session.impl.ReactiveSessionImpl.createReactiveQuery(ReactiveSessionImpl.java:358)\n\tat org.hibernate.reactive.mutiny.impl.MutinySessionImpl.createQuery(MutinySessionImpl.java:129)\n\tat io.quarkus.hibernate.reactive.panache.common.runtime.AbstractJpaOperations.lambda$executeUpdate$30(AbstractJpaOperations.java:405)\n\tat io.smallrye.context.impl.wrappers.SlowContextualFunction.apply(SlowContextualFunction.java:21)\n\tat io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni$UniOnItemTransformToUniProcessor.performInnerSubscription(UniOnItemTransformToUni.java:68)\n\tat io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni$UniOnItemTransformToUniProcessor.onItem(UniOnItemTransformToUni.java:57)\n\tat io.smallrye.mutiny.operators.uni.builders.UniCreateFromKnownItem$KnownItemSubscription.forward(UniCreateFromKnownItem.java:38)\n\tat io.smallrye.mutiny.operators.uni.builders.UniCreateFromKnownItem.subscribe(UniCreateFromKnownItem.java:23)\n\tat io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)\n\tat io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni.subscribe(UniOnItemTransformToUni.java:25)\n\tat io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)\n\tat io.smallrye.mutiny.operators.uni.UniOnFailureTransform.subscribe(UniOnFailureTransform.java:31)\n\tat io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)\n\tat io.smallrye.mutiny.operators.uni.UniOnItemConsume.subscribe(UniOnItemConsume.java:30)\n\tat io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)\n\tat io.smallrye.mutiny.operators.uni.UniOnFailureFlatMap.subscribe(UniOnFailureFlatMap.java:31)\n\tat io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)\n\tat io.smallrye.mutiny.operators.uni.UniOnItemTransform.subscribe(UniOnItemTransform.java:22)\n\tat io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)\n\tat io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni$UniOnItemTransformToUniProcessor.performInnerSubscription(UniOnItemTransformToUni.java:81)\n\tat io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni$UniOnItemTransformToUniProcessor.onItem(UniOnItemTransformToUni.java:57)\n\tat io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni$UniOnItemTransformToUniProcessor.onItem(UniOnItemTransformToUni.java:60)\n\tat io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni$UniOnItemTransformToUniProcessor.onItem(UniOnItemTransformToUni.java:60)\n\tat io.smallrye.mutiny.operators.uni.builders.UniCreateFromKnownItem$KnownItemSubscription.forward(UniCreateFromKnownItem.java:38)\n\tat io.smallrye.mutiny.operators.uni.builders.UniCreateFromKnownItem.subscribe(UniCreateFromKnownItem.java:23)\n\tat io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)\n\tat io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni$UniOnItemTransformToUniProcessor.performInnerSubscription(UniOnItemTransformToUni.java:81)\n\tat io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni$UniOnItemTransformToUniProcessor.onItem(UniOnItemTransformToUni.java:57)\n\tat io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni$UniOnItemTransformToUniProcessor.onItem(UniOnItemTransformToUni.java:60)\n\tat io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni$UniOnItemTransformToUniProcessor.onItem(UniOnItemTransformToUni.java:60)\n\tat io.smallrye.mutiny.operators.uni.UniOperatorProcessor.onItem(UniOperatorProcessor.java:47)\n\tat io.smallrye.mutiny.operators.uni.builders.UniCreateFromCompletionStage$CompletionStageUniSubscription.forwardResult(UniCreateFromCompletionStage.java:63)\n\tat java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:859)\n\tat java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:837)\n\tat java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)\n\tat java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2079)\n\tat org.hibernate.reactive.util.async.impl.AsyncTrampoline$TrampolineInternal.unroll(AsyncTrampoline.java:131)\n\tat org.hibernate.reactive.util.async.impl.AsyncTrampoline$TrampolineInternal.lambda$unroll$0(AsyncTrampoline.java:126)\n\tat java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:859)\n\tat java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:837)\n\tat java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)\n\tat java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2079)\n\tat io.vertx.core.Future.lambda$toCompletionStage$3(Future.java:536)\n\tat io.vertx.core.impl.future.FutureImpl$3.onSuccess(FutureImpl.java:141)\n\tat io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)\n\tat io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)\n\tat io.vertx.core.impl.future.PromiseImpl.tryComplete(PromiseImpl.java:23)\n\tat io.vertx.sqlclient.impl.QueryResultBuilder.tryComplete(QueryResultBuilder.java:88)\n\tat io.vertx.sqlclient.impl.QueryResultBuilder.tryComplete(QueryResultBuilder.java:32)\n\tat io.vertx.core.Promise.complete(Promise.java:66)\n\tat io.vertx.core.Promise.handle(Promise.java:51)\n\tat io.vertx.core.Promise.handle(Promise.java:29)\n\tat io.vertx.core.impl.future.FutureImpl$3.onSuccess(FutureImpl.java:141)\n\tat io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)\n\tat io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)\n\tat io.vertx.core.impl.future.PromiseImpl.tryComplete(PromiseImpl.java:23)\n\tat io.vertx.core.impl.future.PromiseImpl.onSuccess(PromiseImpl.java:49)\n\tat io.vertx.core.impl.future.PromiseImpl.handle(PromiseImpl.java:41)\n\tat io.vertx.sqlclient.impl.TransactionImpl.lambda$wrap$0(TransactionImpl.java:72)\n\tat io.vertx.core.impl.future.FutureImpl$3.onSuccess(FutureImpl.java:141)\n\tat io.vertx.core.impl.future.FutureBase.lambda$emitSuccess$0(FutureBase.java:54)\n\tat io.vertx.core.impl.EventLoopContext.execute(EventLoopContext.java:86)\n\tat io.vertx.core.impl.DuplicatedContext.execute(DuplicatedContext.java:184)\n\tat io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:51)\n\tat io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)\n\tat io.vertx.core.impl.future.PromiseImpl.tryComplete(PromiseImpl.java:23)\n\tat io.vertx.core.impl.future.PromiseImpl.onSuccess(PromiseImpl.java:49)\n\tat io.vertx.core.impl.future.PromiseImpl.handle(PromiseImpl.java:41)\n\tat io.vertx.core.impl.future.PromiseImpl.handle(PromiseImpl.java:23)\n\tat io.vertx.sqlclient.impl.command.CommandResponse.fire(CommandResponse.java:46)\n\tat io.vertx.sqlclient.impl.SocketConnectionBase.handleMessage(SocketConnectionBase.java:325)\n\tat io.vertx.pgclient.impl.PgSocketConnection.handleMessage(PgSocketConnection.java:118)\n\tat io.vertx.sqlclient.impl.SocketConnectionBase.lambda$init$0(SocketConnectionBase.java:138)\n\tat io.vertx.core.impl.EventLoopContext.emit(EventLoopContext.java:55)\n\tat io.vertx.core.impl.ContextBase.emit(ContextBase.java:297)\n\tat io.vertx.core.net.impl.NetSocketImpl.handleMessage(NetSocketImpl.java:378)\n\tat io.vertx.core.net.impl.ConnectionBase.read(ConnectionBase.java:159)\n\tat io.vertx.core.net.impl.VertxHandler.channelRead(VertxHandler.java:153)\n\tat io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)\n\tat io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)\n\tat io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)\n\tat io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:436)\n\tat io.vertx.pgclient.impl.codec.PgEncoder.lambda$write$0(PgEncoder.java:100)\n\tat io.vertx.pgclient.impl.codec.PgCommandCodec.handleReadyForQuery(PgCommandCodec.java:139)\n\tat io.vertx.pgclient.impl.codec.PgDecoder.decodeReadyForQuery(PgDecoder.java:237)\n\tat io.vertx.pgclient.impl.codec.PgDecoder.channelRead(PgDecoder.java:96)\n\tat io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:251)\n\tat io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)\n\tat io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)\n\tat io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)\n\tat io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)\n\tat io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440)\n\tat io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)\n\tat io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)\n\tat io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)\n\tat io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788)\n\tat io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)\n\tat io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)\n\tat io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)\n\tat io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)\n\tat io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)\n\tat io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)\n\tat java.base/java.lang.Thread.run(Thread.java:829)\nCaused by: org.hibernate.query.SyntaxException: At 1:70 and token '<EOF>', mismatched input '<EOF>', expecting one of the following tokens: '=', '.' [UPDATE io.quarkus.it.panache.reactive.Person SET Person.updateAllNames]\n\tat org.hibernate.query.hql.internal.StandardHqlTranslator$1.syntaxError(StandardHqlTranslator.java:108)\n\tat org.antlr.v4.runtime.ProxyErrorListener.syntaxError(ProxyErrorListener.java:41)\n\tat org.antlr.v4.runtime.Parser.notifyErrorListeners(Parser.java:544)\n\tat org.antlr.v4.runtime.DefaultErrorStrategy.reportInputMismatch(DefaultErrorStrategy.java:327)\n\tat org.antlr.v4.runtime.DefaultErrorStrategy.reportError(DefaultErrorStrategy.java:139)\n\tat org.hibernate.grammars.hql.HqlParser.assignment(HqlParser.java:791)\n\tat org.hibernate.grammars.hql.HqlParser.setClause(HqlParser.java:717)\n\tat org.hibernate.grammars.hql.HqlParser.updateStatement(HqlParser.java:651)\n\tat org.hibernate.grammars.hql.HqlParser.statement(HqlParser.java:343)\n\tat org.hibernate.query.hql.internal.StandardHqlTranslator.parseHql(StandardHqlTranslator.java:143)\n\tat org.hibernate.query.hql.internal.StandardHqlTranslator.translate(StandardHqlTranslator.java:67)\n\tat org.hibernate.query.internal.QueryInterpretationCacheStandardImpl.createHqlInterpretation(QueryInterpretationCacheStandardImpl.java:165)\n\tat org.hibernate.query.internal.QueryInterpretationCacheStandardImpl.resolveHqlInterpretation(QueryInterpretationCacheStandardImpl.java:147)\n\tat org.hibernate.internal.AbstractSharedSessionContract.interpretHql(AbstractSharedSessionContract.java:769)\n\tat org.hibernate.reactive.session.impl.ReactiveSessionImpl.createReactiveQuery(ReactiveSessionImpl.java:368)\n\t... 103 more"}
2023-11-24T18:53:35.4999355Z 
2023-11-24T18:53:35.5000079Z 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
2023-11-24T18:53:35.5001814Z 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
2023-11-24T18:53:35.5003659Z 	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
2023-11-24T18:53:35.5005296Z 	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
2023-11-24T18:53:35.5006624Z 	at org.codehaus.groovy.reflection.CachedConstructor.invoke(CachedConstructor.java:73)
2023-11-24T18:53:35.5008220Z 	at org.codehaus.groovy.reflection.CachedConstructor.doConstructorInvoke(CachedConstructor.java:60)
2023-11-24T18:53:35.5010131Z 	at org.codehaus.groovy.runtime.callsite.ConstructorSite$ConstructorSiteNoUnwrap.callConstructor(ConstructorSite.java:86)
2023-11-24T18:53:35.5012067Z 	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallConstructor(CallSiteArray.java:57)
2023-11-24T18:53:35.5014200Z 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:263)
2023-11-24T18:53:35.5016340Z 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:277)
2023-11-24T18:53:35.5018525Z 	at io.restassured.internal.ResponseSpecificationImpl$HamcrestAssertionClosure.validate(ResponseSpecificationImpl.groovy:512)
2023-11-24T18:53:35.5020502Z 	at io.restassured.internal.ResponseSpecificationImpl$HamcrestAssertionClosure$validate$1.call(Unknown Source)
2023-11-24T18:53:35.5022469Z 	at io.restassured.internal.ResponseSpecificationImpl.validateResponseIfRequired(ResponseSpecificationImpl.groovy:696)
2023-11-24T18:53:35.5024586Z 	at io.restassured.internal.ResponseSpecificationImpl.this$2$validateResponseIfRequired(ResponseSpecificationImpl.groovy)
2023-11-24T18:53:35.5026312Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2023-11-24T18:53:35.5027826Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2023-11-24T18:53:35.5029754Z 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2023-11-24T18:53:35.5031163Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2023-11-24T18:53:35.5032638Z 	at org.codehaus.groovy.runtime.callsite.PlainObjectMetaMethodSite.doInvoke(PlainObjectMetaMethodSite.java:43)
2023-11-24T18:53:35.5034741Z 	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:198)
2023-11-24T18:53:35.5036739Z 	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:62)
2023-11-24T18:53:35.5038429Z 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:185)
2023-11-24T18:53:35.5040064Z 	at io.restassured.internal.ResponseSpecificationImpl.body(ResponseSpecificationImpl.groovy:107)
2023-11-24T18:53:35.5041783Z 	at io.restassured.internal.ValidatableResponseOptionsImpl.body(ValidatableResponseOptionsImpl.java:238)
2023-11-24T18:53:35.5043745Z 	at io.quarkus.it.panache.reactive.PanacheFunctionalityTest.testPanacheFunctionality(PanacheFunctionalityTest.java:50)
2023-11-24T18:53:35.5045274Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2023-11-24T18:53:35.5046526Z 	at io.quarkus.test.junit.QuarkusTestExtension.runExtensionMethod(QuarkusTestExtension.java:1013)
2023-11-24T18:53:35.5048117Z 	at io.quarkus.test.junit.QuarkusTestExtension.interceptTestMethod(QuarkusTestExtension.java:827)
2023-11-24T18:53:35.5049424Z 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
2023-11-24T18:53:35.5050369Z 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
2023-11-24T18:53:35.5053277Z 	Suppressed: java.lang.IllegalArgumentException: org.hibernate.query.SyntaxException: At 1:70 and token '<EOF>', mismatched input '<EOF>', expecting one of the following tokens: '=', '.' [UPDATE io.quarkus.it.panache.reactive.Person SET Person.updateAllNames]
2023-11-24T18:53:35.5056497Z 		at org.hibernate.internal.ExceptionConverterImpl.convert(ExceptionConverterImpl.java:143)
2023-11-24T18:53:35.5058006Z 		at org.hibernate.internal.ExceptionConverterImpl.convert(ExceptionConverterImpl.java:167)
2023-11-24T18:53:35.5059488Z 		at org.hibernate.internal.ExceptionConverterImpl.convert(ExceptionConverterImpl.java:173)
2023-11-24T18:53:35.5061227Z 		at org.hibernate.reactive.session.impl.ReactiveExceptionConverter.convert(ReactiveExceptionConverter.java:30)
2023-11-24T18:53:35.5063161Z 		at org.hibernate.reactive.session.impl.ReactiveSessionImpl.createReactiveQuery(ReactiveSessionImpl.java:377)
2023-11-24T18:53:35.5065072Z 		at org.hibernate.reactive.session.impl.ReactiveSessionImpl.createReactiveQuery(ReactiveSessionImpl.java:358)
2023-11-24T18:53:35.5066858Z 		at org.hibernate.reactive.mutiny.impl.MutinySessionImpl.createQuery(MutinySessionImpl.java:129)
2023-11-24T18:53:35.5069035Z 		at io.quarkus.hibernate.reactive.panache.common.runtime.AbstractJpaOperations.lambda$executeUpdate$30(AbstractJpaOperations.java:405)
2023-11-24T18:53:35.5071079Z 		at io.smallrye.context.impl.wrappers.SlowContextualFunction.apply(SlowContextualFunction.java:21)
2023-11-24T18:53:35.5073360Z 		at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni$UniOnItemTransformToUniProcessor.performInnerSubscription(UniOnItemTransformToUni.java:68)
2023-11-24T18:53:35.5075849Z 		at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni$UniOnItemTransformToUniProcessor.onItem(UniOnItemTransformToUni.java:57)
2023-11-24T18:53:35.5078080Z 		at io.smallrye.mutiny.operators.uni.builders.UniCreateFromKnownItem$KnownItemSubscription.forward(UniCreateFromKnownItem.java:38)
2023-11-24T18:53:35.5080136Z 		at io.smallrye.mutiny.operators.uni.builders.UniCreateFromKnownItem.subscribe(UniCreateFromKnownItem.java:23)
2023-11-24T18:53:35.5081778Z 		at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)
2023-11-24T18:53:35.5083302Z 		at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni.subscribe(UniOnItemTransformToUni.java:25)
2023-11-24T18:53:35.5084948Z 		at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)
2023-11-24T18:53:35.5086410Z 		at io.smallrye.mutiny.operators.uni.UniOnFailureTransform.subscribe(UniOnFailureTransform.java:31)
2023-11-24T18:53:35.5087875Z 		at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)
2023-11-24T18:53:35.5089222Z 		at io.smallrye.mutiny.operators.uni.UniOnItemConsume.subscribe(UniOnItemConsume.java:30)
2023-11-24T18:53:35.5090583Z 		at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)
2023-11-24T18:53:35.5091984Z 		at io.smallrye.mutiny.operators.uni.UniOnFailureFlatMap.subscribe(UniOnFailureFlatMap.java:31)
2023-11-24T18:53:35.5093397Z 		at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)
2023-11-24T18:53:35.5094783Z 		at io.smallrye.mutiny.operators.uni.UniOnItemTransform.subscribe(UniOnItemTransform.java:22)
2023-11-24T18:53:35.5096435Z 		at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)
2023-11-24T18:53:35.5098372Z 		at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni$UniOnItemTransformToUniProcessor.performInnerSubscription(UniOnItemTransformToUni.java:81)
2023-11-24T18:53:35.5100835Z 		at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni$UniOnItemTransformToUniProcessor.onItem(UniOnItemTransformToUni.java:57)
2023-11-24T18:53:35.5103086Z 		at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni$UniOnItemTransformToUniProcessor.onItem(UniOnItemTransformToUni.java:60)
2023-11-24T18:53:35.5105316Z 		at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni$UniOnItemTransformToUniProcessor.onItem(UniOnItemTransformToUni.java:60)
2023-11-24T18:53:35.5107562Z 		at io.smallrye.mutiny.operators.uni.builders.UniCreateFromKnownItem$KnownItemSubscription.forward(UniCreateFromKnownItem.java:38)
2023-11-24T18:53:35.5109634Z 		at io.smallrye.mutiny.operators.uni.builders.UniCreateFromKnownItem.subscribe(UniCreateFromKnownItem.java:23)
2023-11-24T18:53:35.5111420Z 		at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)
2023-11-24T18:53:35.5113352Z 		at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni$UniOnItemTransformToUniProcessor.performInnerSubscription(UniOnItemTransformToUni.java:81)
2023-11-24T18:53:35.5115836Z 		at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni$UniOnItemTransformToUniProcessor.onItem(UniOnItemTransformToUni.java:57)
2023-11-24T18:53:35.5118112Z 		at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni$UniOnItemTransformToUniProcessor.onItem(UniOnItemTransformToUni.java:60)
2023-11-24T18:53:35.5120357Z 		at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni$UniOnItemTransformToUniProcessor.onItem(UniOnItemTransformToUni.java:60)
2023-11-24T18:53:35.5122274Z 		at io.smallrye.mutiny.operators.uni.UniOperatorProcessor.onItem(UniOperatorProcessor.java:47)
2023-11-24T18:53:35.5124607Z 		at io.smallrye.mutiny.operators.uni.builders.UniCreateFromCompletionStage$CompletionStageUniSubscription.forwardResult(UniCreateFromCompletionStage.java:63)
2023-11-24T18:53:35.5126819Z 		at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:859)
2023-11-24T18:53:35.5128593Z 		at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:837)
2023-11-24T18:53:35.5130205Z 		at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
2023-11-24T18:53:35.5131694Z 		at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2079)
2023-11-24T18:53:35.5133347Z 		at org.hibernate.reactive.util.async.impl.AsyncTrampoline$TrampolineInternal.unroll(AsyncTrampoline.java:131)
2023-11-24T18:53:35.5135256Z 		at org.hibernate.reactive.util.async.impl.AsyncTrampoline$TrampolineInternal.lambda$unroll$0(AsyncTrampoline.java:126)
2023-11-24T18:53:35.5137239Z 		at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:859)
2023-11-24T18:53:35.5139056Z 		at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:837)
2023-11-24T18:53:35.5140661Z 		at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
2023-11-24T18:53:35.5142156Z 		at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2079)
2023-11-24T18:53:35.5143414Z 		at io.vertx.core.Future.lambda$toCompletionStage$3(Future.java:536)
2023-11-24T18:53:35.5144516Z 		at io.vertx.core.impl.future.FutureImpl$3.onSuccess(FutureImpl.java:141)
2023-11-24T18:53:35.5145673Z 		at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
2023-11-24T18:53:35.5146838Z 		at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
2023-11-24T18:53:35.5148010Z 		at io.vertx.core.impl.future.PromiseImpl.tryComplete(PromiseImpl.java:23)
2023-11-24T18:53:35.5149295Z 		at io.vertx.sqlclient.impl.QueryResultBuilder.tryComplete(QueryResultBuilder.java:88)
2023-11-24T18:53:35.5150701Z 		at io.vertx.sqlclient.impl.QueryResultBuilder.tryComplete(QueryResultBuilder.java:32)
2023-11-24T18:53:35.5151789Z 		at io.vertx.core.Promise.complete(Promise.java:66)
2023-11-24T18:53:35.5152546Z 		at io.vertx.core.Promise.handle(Promise.java:51)
2023-11-24T18:53:35.5153283Z 		at io.vertx.core.Promise.handle(Promise.java:29)
2023-11-24T18:53:35.5154197Z 		at io.vertx.core.impl.future.FutureImpl$3.onSuccess(FutureImpl.java:141)
2023-11-24T18:53:35.5155324Z 		at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
2023-11-24T18:53:35.5156473Z 		at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
2023-11-24T18:53:35.5157642Z 		at io.vertx.core.impl.future.PromiseImpl.tryComplete(PromiseImpl.java:23)
2023-11-24T18:53:35.5158796Z 		at io.vertx.core.impl.future.PromiseImpl.onSuccess(PromiseImpl.java:49)
2023-11-24T18:53:35.5159903Z 		at io.vertx.core.impl.future.PromiseImpl.handle(PromiseImpl.java:41)
2023-11-24T18:53:35.5161216Z 		at io.vertx.sqlclient.impl.TransactionImpl.lambda$wrap$0(TransactionImpl.java:72)
2023-11-24T18:53:35.5162421Z 		at io.vertx.core.impl.future.FutureImpl$3.onSuccess(FutureImpl.java:141)
2023-11-24T18:53:35.5163590Z 		at io.vertx.core.impl.future.FutureBase.lambda$emitSuccess$0(FutureBase.java:54)
2023-11-24T18:53:35.5164767Z 		at io.vertx.core.impl.EventLoopContext.execute(EventLoopContext.java:86)
2023-11-24T18:53:35.5165919Z 		at io.vertx.core.impl.DuplicatedContext.execute(DuplicatedContext.java:184)
2023-11-24T18:53:35.5167092Z 		at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:51)
2023-11-24T18:53:35.5168242Z 		at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
2023-11-24T18:53:35.5169434Z 		at io.vertx.core.impl.future.PromiseImpl.tryComplete(PromiseImpl.java:23)
2023-11-24T18:53:35.5170602Z 		at io.vertx.core.impl.future.PromiseImpl.onSuccess(PromiseImpl.java:49)
2023-11-24T18:53:35.5171715Z 		at io.vertx.core.impl.future.PromiseImpl.handle(PromiseImpl.java:41)
2023-11-24T18:53:35.5172819Z 		at io.vertx.core.impl.future.PromiseImpl.handle(PromiseImpl.java:23)
2023-11-24T18:53:35.5174205Z 		at io.vertx.sqlclient.impl.command.CommandResponse.fire(CommandResponse.java:46)
2023-11-24T18:53:35.5175837Z 		at io.vertx.sqlclient.impl.SocketConnectionBase.handleMessage(SocketConnectionBase.java:325)
2023-11-24T18:53:35.5177558Z 		at io.vertx.pgclient.impl.PgSocketConnection.handleMessage(PgSocketConnection.java:118)
2023-11-24T18:53:35.5179114Z 		at io.vertx.sqlclient.impl.SocketConnectionBase.lambda$init$0(SocketConnectionBase.java:138)
2023-11-24T18:53:35.5180440Z 		at io.vertx.core.impl.EventLoopContext.emit(EventLoopContext.java:55)
2023-11-24T18:53:35.5181540Z 		at io.vertx.core.impl.ContextBase.emit(ContextBase.java:297)
2023-11-24T18:53:35.5182682Z 		at io.vertx.core.net.impl.NetSocketImpl.handleMessage(NetSocketImpl.java:378)
2023-11-24T18:53:35.5183934Z 		at io.vertx.core.net.impl.ConnectionBase.read(ConnectionBase.java:159)
2023-11-24T18:53:35.5185111Z 		at io.vertx.core.net.impl.VertxHandler.channelRead(VertxHandler.java:153)
2023-11-24T18:53:35.5186900Z 		at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
2023-11-24T18:53:35.5188798Z 		at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
2023-11-24T18:53:35.5190672Z 		at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
2023-11-24T18:53:35.5192741Z 		at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:436)
2023-11-24T18:53:35.5194529Z 		at io.vertx.pgclient.impl.codec.PgEncoder.lambda$write$0(PgEncoder.java:100)
2023-11-24T18:53:35.5195947Z 		at io.vertx.pgclient.impl.codec.PgCommandCodec.handleReadyForQuery(PgCommandCodec.java:139)
2023-11-24T18:53:35.5197437Z 		at io.vertx.pgclient.impl.codec.PgDecoder.decodeReadyForQuery(PgDecoder.java:237)
2023-11-24T18:53:35.5198744Z 		at io.vertx.pgclient.impl.codec.PgDecoder.channelRead(PgDecoder.java:96)
2023-11-24T18:53:35.5200216Z 		at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:251)
2023-11-24T18:53:35.5201973Z 		at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
2023-11-24T18:53:35.5203820Z 		at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
2023-11-24T18:53:35.5205507Z 		at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
2023-11-24T18:53:35.5206498Z 		at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
2023-11-24T18:53:35.5207469Z 		at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440)
2023-11-24T18:53:35.5208505Z 		at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
2023-11-24T18:53:35.5209604Z 		at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
2023-11-24T18:53:35.5210484Z 		at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
2023-11-24T18:53:35.5211295Z 		at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788)
2023-11-24T18:53:35.5212085Z 		at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)
2023-11-24T18:53:35.5213083Z 		at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)
2023-11-24T18:53:35.5213742Z 		at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
2023-11-24T18:53:35.5214468Z 		at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
2023-11-24T18:53:35.5215295Z 		at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
2023-11-24T18:53:35.5216297Z 		at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
2023-11-24T18:53:35.5217055Z 		at java.base/java.lang.Thread.run(Thread.java:829)
2023-11-24T18:53:35.5218563Z 	Caused by: org.hibernate.query.SyntaxException: At 1:70 and token '<EOF>', mismatched input '<EOF>', expecting one of the following tokens: '=', '.' [UPDATE io.quarkus.it.panache.reactive.Person SET Person.updateAllNames]
2023-11-24T18:53:35.5219935Z 		at org.hibernate.query.hql.internal.StandardHqlTranslator$1.syntaxError(StandardHqlTranslator.java:108)
2023-11-24T18:53:35.5220797Z 		at org.antlr.v4.runtime.ProxyErrorListener.syntaxError(ProxyErrorListener.java:41)
2023-11-24T18:53:35.5221506Z 		at org.antlr.v4.runtime.Parser.notifyErrorListeners(Parser.java:544)
2023-11-24T18:53:35.5222272Z 		at org.antlr.v4.runtime.DefaultErrorStrategy.reportInputMismatch(DefaultErrorStrategy.java:327)
2023-11-24T18:53:35.5223127Z 		at org.antlr.v4.runtime.DefaultErrorStrategy.reportError(DefaultErrorStrategy.java:139)
2023-11-24T18:53:35.5223857Z 		at org.hibernate.grammars.hql.HqlParser.assignment(HqlParser.java:791)
2023-11-24T18:53:35.5224585Z 		at org.hibernate.grammars.hql.HqlParser.setClause(HqlParser.java:717)
2023-11-24T18:53:35.5225253Z 		at org.hibernate.grammars.hql.HqlParser.updateStatement(HqlParser.java:651)
2023-11-24T18:53:35.5225923Z 		at org.hibernate.grammars.hql.HqlParser.statement(HqlParser.java:343)
2023-11-24T18:53:35.5226712Z 		at org.hibernate.query.hql.internal.StandardHqlTranslator.parseHql(StandardHqlTranslator.java:143)
2023-11-24T18:53:35.5227651Z 		at org.hibernate.query.hql.internal.StandardHqlTranslator.translate(StandardHqlTranslator.java:67)
2023-11-24T18:53:35.5228842Z 		at org.hibernate.query.internal.QueryInterpretationCacheStandardImpl.createHqlInterpretation(QueryInterpretationCacheStandardImpl.java:165)
2023-11-24T18:53:35.5230420Z 		at org.hibernate.query.internal.QueryInterpretationCacheStandardImpl.resolveHqlInterpretation(QueryInterpretationCacheStandardImpl.java:147)
2023-11-24T18:53:35.5231666Z 		at org.hibernate.internal.AbstractSharedSessionContract.interpretHql(AbstractSharedSessionContract.java:769)
2023-11-24T18:53:35.5232725Z 		at org.hibernate.reactive.session.impl.ReactiveSessionImpl.createReactiveQuery(ReactiveSessionImpl.java:368)
2023-11-24T18:53:35.5233405Z 		... 103 more
```

DavideD commented:
It seems that the update query is missing the parameter bit, the update query should end with ` := $1`  or something.

My guess is that this is an error in Panache

DavideD commented:
It's probably something simpler than that. I will have a look

DavideD committed:
Add SyntaxException to the NamedQueryMistake checks

For Panache ORM and Reactive

DavideD committed:
Add test for Panache ORM query exception

DavideD commented:
Hibernate ORM now throws a different error that's ignored by Panache. This is true for Hibernate ORM too.
But, there's no similar test for Hibernate ORM and so there's no failure.

A fix like this should work: https://github.com/quarkusio/quarkus/pull/37338/commits/86d0040c5e066c970d0acdc0fe18047929dac3c8

I've added three commits on top of this PR here: https://github.com/quarkusio/quarkus/pull/37338

yrodiere commented:
> Hibernate ORM now throws a different error that's ignored by Panache. This is true for Hibernate ORM too. But, there's no similar test for Hibernate ORM and so there's no failure.
> 
> A fix like this should work: [86d0040](https://github.com/quarkusio/quarkus/commit/86d0040c5e066c970d0acdc0fe18047929dac3c8)
> 
> I've added three commits on top of this PR here: #37338

Thanks! I added your commits on top of this PR. Let's run CI again!

yrodiere commented:
@DavideD I think your cleanup commit actually breaks some tests... It looks like `Person::count` is actually compiled as `AbstractJpaOperations::count`, resulting in a runtime failure? @FroMage do you know anything about this?

Anyway, I'll remove that commit. Feel free to open a separate PR about it @DavideD , but I suspect this will also need a fix in Panache (if Panache can even do something about it).

yrodiere commented:
Alright, the build finally passed! :partying_face: 

Could I get an approval to get this merged, @gsmet?
Or @Sanne, but I think he's busy these days.

gsmet reviewed (changes requested):
We will need a note in the migration guide about:

> Rename extension quarkus-hibernate-search-orm-coordination-outbox-polling to quarkus-hibernate-search-orm-outbox-polling

And if it's an extension that can be consumed by users, we should:
- keep it defined in the BOM for a while
- have redirections in place in the relocations/ module. They should be short given only the artifact changes.

This can be done once the PR is merged, I will merge it.

gsmet reviewed (approved):
Ah, I wanted to post an Approval review, and apparently clicked on the wrong checkbox.

As said, what I wrote above is not blocking, will merge!

gsmet commented:
Thanks everyone, nice to have the latest ORM in Quarkus!

yrodiere commented:
> We will need a note in the migration guide about:

I added relevant notes for both ORM and Search to https://github.com/quarkusio/quarkus/wiki/Migration-Guide-3.7

> And if it's an extension that can be consumed by users, we should:
>
>    keep it defined in the BOM for a while
>    have redirections in place in the relocations/ module. They should be short given only the artifact changes.

I kept it in the BOM. Will submit a PR for relocations.
